src/nvim/option.c

```diff
@@ -301,6 +301,15 @@ static char *(p_cot_values[]) = {"menu", "menuone", "longest", "preview",
 # include "option.c.generated.h"
 #endif
 
+static void set_runtimepath_default(void)
+{
+  garray_T rtp_ga;
+  ga_init(&rtp_ga, (int)sizeof(const char *), 1);
+  GA_APPEND(const char *, &rtp_ga, get_user_conf_dir());
+  GA_APPEND(const char *, &rtp_ga, concat_fnames(get_user_conf_dir(), "after", true));
+  set_string_default("runtimepath", ga_concat_strings(&rtp_ga));
```

Comment:
Relying on `PATH_MAX` is extremely silly and not a very well researched suggestion.  [Consider](http://insanecoding.blogspot.co.uk/2007/11/pathmax-simply-isnt.html) the implications of what you're suggesting on a portability level.  No, correct handling of paths involve using calloc and snprintf.  (I assume xmalloc is just a wrapper around malloc that provides overflow checking like calloc but without the guarantee of zero initialised memory.)